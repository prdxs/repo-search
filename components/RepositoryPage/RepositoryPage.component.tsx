import React from 'react';
import clsx from 'clsx';
import styled from 'styled-components';
import Grid from '@material-ui/core/Grid';

import RepostoryCard from '@/components/RepositoryCard';
import { IRepositoryPageProps } from './RepositoryPage.typings';

const RepositoryPage: React.FC<IRepositoryPageProps> = ({
  className,
  style,
  name,
  link,
  stars,
  forks,
  issues,
  lastUpdated,
}) => (
  <div className={clsx('RepositoryPage-root', className)} style={style}>
    <Grid container justify="center" alignItems="center">
      <Grid item>
        <RepostoryCard
          name={name}
          link={link}
          stars={stars}
          forks={forks}
          issues={issues}
          lastUpdated={lastUpdated}
        />
      </Grid>
    </Grid>
  </div>
);

const StyledRepositoryPage = styled(RepositoryPage)`
  &.RepositoryPage-root {
    height: 100vh;

    > .MuiGrid-container:first-child {
      padding-top: ${({ theme }) => theme.spacing(8)};
      padding-bottom: ${({ theme }) => theme.spacing(16)};
    }
  }
`;

export default StyledRepositoryPage;
