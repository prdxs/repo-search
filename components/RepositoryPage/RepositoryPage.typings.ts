import { IComponentProps } from '@/typings/common';
import { IIssue } from './state/typings';

export interface IRepositoryPageProps extends IComponentProps {
  name: string;
  link: string;
  stars: number;
  forks: number;
  issues: number;
  lastUpdated: string;
  issueList: Array<IIssue>;
  loadingIssues: boolean;
}
