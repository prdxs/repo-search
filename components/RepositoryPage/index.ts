export { default } from './RepositoryPage.component';
