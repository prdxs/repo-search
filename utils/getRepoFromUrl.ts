const getRepoFromUrl = (url: string): { owner: string; repo: string } => {
  const parts = url.split('/');
  const len = parts.length;
  const owner = parts[len - 2];
  const repo = parts[len - 1];

  return { owner, repo };
};

export default getRepoFromUrl;
