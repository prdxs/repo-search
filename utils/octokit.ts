import { Octokit } from '@octokit/rest';

const octokit = new Octokit({
  baseUrl: 'https://api.github.com',
  auth: process.env.NEXT_PUBLIC_GH_TOKEN,
});

export default octokit;
