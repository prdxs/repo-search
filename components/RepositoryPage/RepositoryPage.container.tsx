import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Grid from '@material-ui/core/Grid';
import CircularProgress from '@material-ui/core/CircularProgress';

import { getRepository } from '@/components/RepositorySearchPage/state/actions';
import { selectRepositoryEntities } from '@/components/RepositorySearchPage/state/selectors';
import RepositoryPage from './RepositoryPage.component';
import { getRepoIssues } from './state/actions';
import { selectAll, selectIssuesLoading } from './state/selectors';

const RepositoryPageContainer: React.FC<{
  id: string;
  owner: string;
  repo: string;
}> = ({ id, owner, repo }) => {
  const dispatch = useDispatch();
  const repositoryDictionary = useSelector(selectRepositoryEntities);
  const issueList = useSelector(selectAll);
  const loadingIssues = useSelector(selectIssuesLoading);

  const selectedRepo = repositoryDictionary[id];

  useEffect(() => {
    dispatch(getRepository({ owner, repo }));
    dispatch(getRepoIssues({ owner, repo }));
  }, [owner, repo]);

  if (selectedRepo === undefined) {
    return (
      <Grid
        style={{ height: '100vh' }}
        container
        justify="center"
        alignItems="center"
      >
        <Grid item>
          <CircularProgress />
        </Grid>
      </Grid>
    );
  }

  return (
    <RepositoryPage
      issueList={issueList}
      loadingIssues={loadingIssues}
      {...selectedRepo}
    />
  );
};

export default RepositoryPageContainer;
