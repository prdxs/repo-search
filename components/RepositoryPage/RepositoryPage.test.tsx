import React from 'react';
import { render } from '@/test/utils';

import RepositoryPage from './RepositoryPage.component';

describe('RepositoryPage', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <RepositoryPage
        name="prdxs/repo-search"
        link="https://github.com/prdxs/repo-search"
        stars={12}
        forks={10}
        issues={100}
        lastUpdated="1st Jul 2020 07:20:59"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
