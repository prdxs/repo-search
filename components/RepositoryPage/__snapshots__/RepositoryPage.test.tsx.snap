// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryPage should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    style="padding: 2rem;"
  >
    <div
      class="RepositoryPage-root RepositoryPagecomponent__StyledRepositoryPage-sc-1acltcm-0 kWxLoC"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root MuiCard-root RepositoryCard-root RepositoryCardcomponent__StyledRepositoryCard-sc-1c2lx1b-0 ljLEiR MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="RepositoryCard-root"
          >
            <div
              class="MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18 2H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 4h5v8l-2.5-1.5L6 12V4z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6"
                      >
                        <a
                          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorSecondary"
                          href="https://github.com/prdxs/repo-search"
                          target="_blank"
                        >
                          prdxs/repo-search
                        </a>
                      </h6>
                    </div>
                  </div>
                </span>
              </div>
            </div>
            <div
              class="MuiCardContent-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <span
                        class="MuiBadge-root"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root RepositoryCard-star MuiSvgIcon-fontSizeLarge"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                          />
                        </svg>
                        <span
                          class="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-colorPrimary"
                        >
                          12
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <span
                        class="MuiBadge-root"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20V8zm-6 8h-4v-2h4v2zm0-4h-4v-2h4v2z"
                          />
                        </svg>
                        <span
                          class="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-colorSecondary"
                        >
                          100
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <span
                        class="MuiBadge-root"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M14 4l2.29 2.29-2.88 2.88 1.42 1.42 2.88-2.88L20 10V4zm-4 0H4v6l2.29-2.29 4.71 4.7V20h2v-8.41l-5.29-5.3z"
                          />
                        </svg>
                        <span
                          class="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-colorSecondary"
                        >
                          10
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <span
                class="MuiTypography-root RepositoryCard-lastUpdated MuiTypography-caption"
              >
                1st Jul 2020 07:20:59
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root RepositoryPage-list MuiGrid-item"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-direction-xs-column"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiPaper-root MuiCard-root IssueCard-root IssueCardcomponent__StyledIssueCard-sc-1lg5wpy-0 etfYhE MuiPaper-elevation1 MuiPaper-rounded"
                data-testid="IssueCard-root"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-avatar"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorError"
                      focusable="false"
                      title="Issue Open"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-6 15h-2v-2h2v2zm0-4h-2V8h2v6zm-1-9c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiCardHeader-content"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-h6"
                    >
                      <a
                        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorSecondary"
                        href="https://github.com/facebook/react/issues/19245"
                        target="_blank"
                      >
                        Bug: what's sky-lighter.appspot.com?
                      </a>
                    </h6>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiPaper-root MuiCard-root IssueCard-root IssueCardcomponent__StyledIssueCard-sc-1lg5wpy-0 etfYhE MuiPaper-elevation1 MuiPaper-rounded"
                data-testid="IssueCard-root"
              >
                <div
                  class="MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-avatar"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-colorAction"
                      focusable="false"
                      title="Issue Closed"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm-2 14l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiCardHeader-content"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-h6"
                    >
                      <a
                        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorSecondary"
                        href="https://github.com/facebook/react/issues/19243"
                        target="_blank"
                      >
                        Bug: Cannot read property 'references' of undefined in eslint-plugin-react-hooks v4.0.5
                      </a>
                    </h6>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
